src/main/scala/edu/ie3/simona/model/participant/HpModel.scala

```diff
@@ -182,69 +179,56 @@ final case class HpModel(
-      currentThermalGridState,
-      demandHouse,
-      demandThermalStorage,
+    val demandHouse = thermalDemands.houseDemand
+    val demandThermalStorage = thermalDemands.heatStorageDemand
+
+    val noThermalStorageOrThermalStorageIsEmpty = determineThermalStorageStatus(
+      currentThermalGridState
     )
 
-    val turnHpOn: Boolean =
-      houseHasDemand || heatStorageHasDemand
+    val turnHpOn =
+      (demandHouse.hasRequiredDemand && noThermalStorageOrThermalStorageIsEmpty) ||
+        (demandHouse.hasAdditionalDemand && lastState.isRunning) ||
+        demandThermalStorage.hasRequiredDemand ||
+        (demandThermalStorage.hasAdditionalDemand && lastState.isRunning)
 
     val canOperate =
       demandHouse.hasRequiredDemand || demandHouse.hasAdditionalDemand ||
         demandThermalStorage.hasRequiredDemand || demandThermalStorage.hasAdditionalDemand
     val canBeOutOfOperation =
       !(demandHouse.hasRequiredDemand && noThermalStorageOrThermalStorageIsEmpty)
 
-    (turnHpOn, canOperate, canBeOutOfOperation)
+    (
+      turnHpOn,
+      canOperate,
+      canBeOutOfOperation,
+    )
   }
 
   /** This method will return booleans whether there is a heat demand of house
     * or thermal storage as well as a boolean indicating if there is no thermal
     * storage, or it is empty.
     *
-    * @param lastHpState
-    *   Current state of the heat pump
     * @param updatedGridState
     *   The updated state of the [[ThermalGrid]]
-    * @param demandHouse
-    *   heat demand of the thermal house
-    * @param demandThermalStorage
-    *   heat demand of the thermal storage
     * @return
-    *   First boolean is true, if house has heat demand. Second boolean is true,
-    *   if thermalStorage has heat demand. Third boolean is true, if there is no
-    *   thermalStorage, or it's empty.
+    *   boolean which is true, if there is no thermalStorage, or it's empty.
     */
 
-  private def determineDemandBooleans(
-      lastHpState: HpState,
-      updatedGridState: ThermalGridState,
-      demandHouse: ThermalEnergyDemand,
-      demandThermalStorage: ThermalEnergyDemand,
-  ): (Boolean, Boolean, Boolean) = {
+  private def determineThermalStorageStatus(
+      updatedGridState: ThermalGridState
+  ): Boolean = {
     implicit val tolerance: Energy = KilowattHours(1e-3)
     val noThermalStorageOrThermalStorageIsEmpty: Boolean =
       updatedGridState.storageState.isEmpty || updatedGridState.storageState
         .exists(
           _.storedEnergy =~ zeroKWh
         )
 
-    val houseDemand =
-      (demandHouse.hasRequiredDemand && noThermalStorageOrThermalStorageIsEmpty) || (lastHpState.isRunning && demandHouse.hasAdditionalDemand)
-    val heatStorageDemand =
-      demandThermalStorage.hasRequiredDemand || (lastHpState.isRunning && demandThermalStorage.hasAdditionalDemand)
-    (houseDemand, heatStorageDemand, noThermalStorageOrThermalStorageIsEmpty)
+    noThermalStorageOrThermalStorageIsEmpty
```

Comment:
You can return the result directly instead of creating a new variable.

Reply:
Good catch!